src/linear_algebra/matrix/hermitian.lean

```diff
@@ -27,56 +29,27 @@ open_locale matrix
 
 local notation `⟪`x`, `y`⟫` := @inner α (pi_Lp 2 (λ (_ : n), α)) _ x y
 
-section non_unital_semiring
-
-variables [non_unital_semiring α] [star_ring α] [non_unital_semiring β] [star_ring β]
+section has_star
+variables [has_star α] [has_star β]
 
 /-- A matrix is hermitian if it is equal to its conjugate transpose. On the reals, this definition
 captures symmetric matrices. -/
 def is_hermitian (A : matrix n n α) : Prop := Aᴴ = A
 
 lemma is_hermitian.eq {A : matrix n n α} (h : A.is_hermitian) : Aᴴ = A := h
 
+lemma is_hermitian.is_self_adjoint {A : matrix n n α} (h : A.is_hermitian) : is_self_adjoint A := h
```

Comment:
This is the only new lemma in this file; the others have been reordered as part of typeclass relaxation.

Reply:
(this is no longer the case after @j-loreaux's comments)